src/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURL.java

```diff
@@ -147,88 +152,56 @@ public void scanHttpRequestSend(HttpMessage msg, int id) {
      */
     @Override
     public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
-        // Simplify everything using only getQuery()???
-        String uri = msg.getRequestHeader().getURI().toString();
-        String sessionIdPair;
-        String sessionIdValue;
-        String sessionIdName;
-        Pattern pattern;
-        Matcher matcher;
 
-        // Alternative implementation
-        // we search for all params looking for one session element
-        // maybe more efficient because we don't build Patterns any time
-        //String[] params = msg.getParamNames();
+        String uri = msg.getRequestHeader().getURI().toString();

```

Comment:
The conversion to `String` could be delayed, to after the `urlParams` size check?